src/Microsoft.Azure.NotificationHubs/KeyValueConfigurationManager.cs

```diff
@@ -106,10 +122,208 @@ private static NameValueCollection CreateNameValueCollectionFromConnectionString
+            string oauthUser = connectionProperty8;
+            SecureString oauthPassword2 = oauthPassword1;
+            return CreateTokenProvider((IEnumerable<Uri>)endpointAddresses,
+                issuerName,
+                issuerKey,
+                sharedAccessKeyName,
+                sharedAccessKey,
+                windowsDomain,
+                windowsUser,
+                windowsPassword2,
+                oauthDomain,
+                oauthUser,
+                oauthPassword2);
+        }
+
+        private static TokenProvider CreateTokenProvider(
+            IEnumerable<Uri> stsEndpoints,
+            string issuerName,
+            string issuerKey,
+            string sharedAccessKeyName,
+            string sharedAccessKey,
+            string windowsDomain,
+            string windowsUser,
+            SecureString windowsPassword,
+            string oauthDomain,
+            string oauthUser,
+            SecureString oauthPassword)
+        {
+            if (!string.IsNullOrWhiteSpace(sharedAccessKey))
+            {
+                return TokenProvider.CreateSharedAccessSignatureTokenProvider(sharedAccessKeyName, sharedAccessKey);
+            }
+            if (string.IsNullOrWhiteSpace(issuerName))
+            {
+                int num = stsEndpoints == null ? 0 : (stsEndpoints.Any<Uri>() ? 1 : 0);
+                bool flag1 = !string.IsNullOrWhiteSpace(windowsUser) && windowsPassword != null && windowsPassword.Length > 0;
+                bool flag2 = !string.IsNullOrWhiteSpace(oauthUser) && oauthPassword != null && oauthPassword.Length > 0;
+                if (num == 0)
+                    return (TokenProvider)null;
+                if (flag2)
+                {
+                    NetworkCredential credential = string.IsNullOrWhiteSpace(oauthDomain)
+                        ? new NetworkCredential(oauthUser, oauthPassword)
+                        : new NetworkCredential(oauthUser, oauthPassword, oauthDomain);
+                    return TokenProvider.CreateOAuthTokenProvider(stsEndpoints, credential);
+                }
+                if (!flag1)
+                {
+                    return TokenProvider.CreateWindowsTokenProvider(stsEndpoints);
+                }
+                NetworkCredential credential1 = string.IsNullOrWhiteSpace(windowsDomain)
+                    ? new NetworkCredential(windowsUser, windowsPassword)
+                    : new NetworkCredential(windowsUser, windowsPassword, windowsDomain);
+                return TokenProvider.CreateWindowsTokenProvider(stsEndpoints, credential1);
+            }
+
+            if (stsEndpoints != null && stsEndpoints.Any<Uri>())
+            {
+                return TokenProvider.CreateSharedSecretTokenProvider(issuerName, issuerKey, stsEndpoints.First<Uri>());
+            }
+            return TokenProvider.CreateSharedSecretTokenProvider(issuerName, issuerKey);
+        }
+
+        private IList<Uri> GetEndpointAddresses()
```

Comment:
This doesn't seem to be used.  Is that intentional?

Reply:
removed